linter_api/src/ast/generic/binding_arg.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{
+    ast::{ty::TyKind, Span, SpanId, SymbolId},
+    context::with_cx,
+    ffi::FfiOption,
+};
+
+#[repr(C)]
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub struct TyBinding<'ast> {
```

Comment:
This should have some doc comment, with an example. The name is not very explanatory